lib/libvfio-user.c

```diff
@@ -553,19 +575,22 @@ handle_dma_map_or_unmap(vfu_ctx_t *vfu_ctx, uint32_t size, bool map,
                                             fd,
                                             dma_regions[i].offset);
             if (ret < 0) {
+                if (fd != -1) {
```

Comment:
Sorry, I'm not sure why we're doing this. What's the point of restore_fd()?
We now own the fd here, so we can and should just close() it.

Reply:
Right, the `restore_fd` logic is unnecessary.